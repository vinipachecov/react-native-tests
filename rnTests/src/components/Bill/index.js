import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

import { theme } from '../../theme';

const Bill = ({ bill, payBill }) => (
  <View style={billStyles.view}>
    <View style={billStyles.inlineGroup}>
      <Text style={billStyles.label}>{bill.label}</Text>
      <Text>
        {`Due on  ${bill.dueDay}`}
      </Text>
    </View>
    {payBill && (
      <TouchableOpacity
        onPress={() => payBill(bill)}
        style={billStyles.payBill}
      >
        <Text>Pay</Text>
      </TouchableOpacity>
    )}
  </View>
);

Bill.propTypes = {
  bill: PropTypes.object.isRequired,
  payBill: PropTypes.func,
};

Bill.propTypes = {
  payBill: () => null,
};

const billStyles = StyleSheet.create({
  view: {
    display: 'flex',
    padding: theme.spaces[1],
    width: '100%',
  },
  inlineGroup: {
    alignItems: 'center',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
  label: {
    fontSize: theme.fontSizes.headerTwo,
  },
  payBill: {
    backgroundColor: theme.colors.accent.default,
    padding: theme.spaces[1],
    width: 35,
  },
});

export default Bill;
