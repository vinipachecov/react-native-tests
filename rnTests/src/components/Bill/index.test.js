import React from 'react';
import { create } from 'react-test-renderer';

import Bill from './';

describe('[Component] Bill', () => {

  const props = {
    bill: {
      dueDay: '11',
      label: 'Rent'
    },
    payBill: jest.fn()
  }
  const tree = create(<Bill { ...props } />);


  it('Matches snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  })
    
})
