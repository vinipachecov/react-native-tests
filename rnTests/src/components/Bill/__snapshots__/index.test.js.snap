// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Bill Matches snapshot 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "padding": 4,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Rent
    </Text>
    <Text>
      Due on  11
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
        "opacity": 1,
        "padding": 4,
        "width": 35,
      }
    }
  >
    <Text>
      Pay
    </Text>
  </View>
</View>
`;
